mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6255,10 +6295,70 @@ struct ThreadwiseCopyRewritePattern
         else
           destLowerIndices = destUpperIndices;
 
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        if (destLowerIndices.size() == 5) {
+          for (unsigned iter = 0; iter < destTransform.getNumResults();
+               ++iter) {
+            auto expr = destTransform.getResult(iter);
+            if (iter == 2) { // put gemmM or gemmN dim
+              toEmitOOBStoreCheckLogic = true;
+              oobStoreCheckDims.push_back(iter);
+            }
+          }
+        }
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          Value withinStoreBoundsOp =
+              b.create<ConstantIntOp>(loc, 1, b.getIntegerType(1));
+
+          auto iv0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
+                                           b.getIntegerType(32));
+          auto iv1 = b.create<IndexCastOp>(loc, destLowerIndices[1],
+                                           b.getIntegerType(32));
+          auto iv2 = b.create<IndexCastOp>(loc, destLowerIndices[2],
+                                           b.getIntegerType(32));
+
+          Value oobAddrOp =
+              b.create<ConstantIntOp>(loc, 2147483648, b.getIntegerType(32));
```

Comment:
per discussion, we don't need this Value here anymore. it would sit inside GPU->ROCDL conversion process instead.

Reply:
we still need it , I pass it as offset